provisioning/transport/mqtt/devdoc/mqtt_requirements.md

```diff
@@ -0,0 +1,127 @@
+# azure-iot-provisioning-device-mqtt
+
+The `Mqtt` and `MqttWs` classes provides the Provisioning Device SDK with a transport for x509 registration flows over Mqtt. It is instantiated by SDK users only to be passed to the `ProvisioningDeviceClient` object and does not have a public API.
+
+# Internal API
+
+These methods are used by the other objects of the SDK but are not public API for the SDK user to call.
+
+## setAuthentication(auth: X509): void
+
+**SRS_NODE_PROVISIONING_MQTT_18_001: [** The certificate and key passed as properties of the `auth` function shall be used to connect to the Device Provisioning Service. **]**
+
+## registrationRequest(request: RegistrationRequest, callback: (err?: Error, responseBody?: any, result?: any, pollingInterval?: number) => void): void
+
+**SRS_NODE_PROVISIONING_MQTT_18_002: [** If the transport is not connected, `registrationRequest` shall connect it and subscribe to the response topic. **]**
+
+**SRS_NODE_PROVISIONING_MQTT_18_003: [** `registrationRequest` shall publish to '$dps/registrations/PUT/iotdps-register/?$rid<rid>'. **]**
+
+**SRS_NODE_PROVISIONING_MQTT_18_004: [** If the publish fails, `registrationRequest` shall wrap the error in a `TransportSpecificError` and call `callback` passing the wrapped error back. **]**
+
+**SRS_NODE_PROVISIONING_MQTT_18_005: [** If the publish fails, `registrationRequest` shall disconnect the transport. **]**
+
+**SRS_NODE_PROVISIONING_MQTT_18_006: [** `registrationRequest` shall wait for a response with a matching rid. **]**
+
+**SRS_NODE_PROVISIONING_MQTT_18_007: [** `registrationRequest` shall wait for a response for `timeoutInterval` milliseconds.  After that shall be considered a timeout. **]**
+
+**SRS_NODE_PROVISIONING_MQTT_18_008: [** When `registrationRequest` times out, it shall disconnect the transport. **]**
+
+**SRS_NODE_PROVISIONING_MQTT_18_009: [** When `registrationRequest` times out, it shall call callback passing in a ServiceUnavailableError. **]**
+
+**SRS_NODE_PROVISIONING_MQTT_18_010: [** When waiting for responses, `registrationRequest` shall watch for messages with a topic named $dps/registrations/res/<status>/?$rid=<rid>. **]**
+
+**SRS_NODE_PROVISIONING_MQTT_18_011: [** When waiting for responses, `registrationRequest` shall ignore any messages that don't match the required topic format. **]**
```

Comment:
>shall ignore any messages that don't match the required topic format. [](start = 91, length = 69)

I don't see how that could happen since we're subscribing specifically to that topic?